scripts/monitoring/cron-haproxy-close-wait.py

```diff
@@ -49,34 +48,45 @@ def kill(self):
         self.current_time = time.time()
         haproxy_procs_etimes = {}
 
+        # get all haproxy processes
         for proc in psutil.process_iter():
             try:
                 if proc.name() == 'haproxy':
-                    for conn in proc.connections():
-                        if conn.status == 'CLOSE_WAIT':
-                            elapsed = self.get_etimes(proc)
-                            self.dprint("PID: {} elapsed: {}".format(proc.pid, elapsed))
-                            if elapsed > self.args.max_elapsed_time:
-                                haproxy_procs_etimes[elapsed] = proc.pid
+                    elapsed = self.get_etimes(proc)
+                    haproxy_procs_etimes[elapsed] = proc.pid
 
             except psutil.NoSuchProcess:
                 pass
 
+        # identify most recent haproxy process
+        # and remove it from list of haproxy processes
         try:
-            newest_proc = min(haproxy_procs_etimes.keys())
-            haproxy_procs_etimes.pop(newest_proc)
+            youngest_etimes = min(haproxy_procs_etimes.keys())
+            youngest_pid = haproxy_procs_etimes[youngest_etimes]
+            self.dprint("Youngest haproxy PID: {}".format(youngest_pid))
+            haproxy_procs_etimes.pop(youngest_etimes)
         except ValueError:
             pass
 
+        # stop any haproxy process older than the most recent
+        # that has connections in 'CLOSE-WAIT' state
+        kill_count = 0
         for proc in haproxy_procs_etimes.values():
             try:
+                kill = False
                 process = psutil.Process(proc)
-                self.dprint("Stopping PID: {}".format(process.pid))
-                process.kill()
+                for conn in process.connections():
+                    if conn.status == 'CLOSE_WAIT':
```

Comment:
This logic still isn't right.  There characteristic we're looking for is a process which has *only* connections (usually just one) in CLOSE_WAIT, and no other states.  Maybe it would be easier to invert it, i.e., pop everything with a connection *other* than CLOSE_WAIT off of the list, and kill the rest.